adoc/chapters/architecture.adoc

```diff
@@ -1072,6 +1071,10 @@ values:
     access the memory allocation containing the referenced object, as
     defined by the capabilities of <<buffer,buffers>> and <<usm>>.
 
+The complete set of memory scopes is not guaranteed to be supported by every
```

Comment:
Should this paragraph also have the "nor across all combinations of devices within a context", to account for the case where some devices in a platform may support system scope, where as another might be more limited?

Reply:
> I think the current spec wording says that the application is not required to check whether D2 supports memory_scope::system. Instead, I think the implementation is required to support atomic synchronization between D1 and D2 because D1 reported memory_scope::system:

So yeah this was the case that I had in mind, I can imagine a theoretical platform where there are multiple devices, say D1 and D2 which provide `mermory_scope::system` atomic synchronization, in that both D1 and D2 can access and modify some object in shared memory concurrently with the host, but not with each other.

Furthermore, this could be extended to a platform where there are subsets of devices that share the same memory scope for synchronization, but not all, say D1, D2, and D3, where D1 and D2 provide `mermory_scope::system` atomic synchronization w.r.t each other but not w.r.t D3.

Obviously, this is all theoretical, and I'm not aware of any specific architectures where this is the case, but I wonder if we need to account for this. I think we perhaps should revise the concept of system-level synchronization such that it doesn't implicitly refer to synchronization w.r.t all other devices in the platform, and instead refers to the relationships between a group of devices and with the host.

I think the wording you suggest makes sense, though how do we define "the atomic operation is to memory that the device can concurrently access"? We currently don't currently specify device-to-device access and I don't think we can use the context as the basis for this, as the criteria for devices being in the same context is too broad.

As a side, I didn't mean to derail this PR, if you'd like I'm happy to leave the wording as you have it and open a new issue to continue this.

Reply:
> As a side, I didn't mean to derail this PR, if you'd like I'm happy to leave the wording as you have it and open a new issue to continue this.

Yes, I think this is better.  In order to solve the problem you are pointing out, we would need to add some new peer-to-peer API that queries about atomic access between arbitrary devices.  This is beyond the scope of the clarification I want to make in this PR.

Reply:
I've created a new issue for this - https://github.com/KhronosGroup/SYCL-Docs/issues/429.